### PR TITLE
LPS-171159 search-experiences-web: Add Logic to return specific error when user try to import a duplicated ERC

### PR DESCRIPTION
**Pull request by gustavolimav**
https://issues.liferay.com/browse/LPS-171159

### CODE REVIEW

**Comment by joshuacords** [comment]
Hi @gustavolimav, this looks pretty good - could you explain how the DuplicateSXPBlueprintExternalReferenceCodeException gets checked for and then thrown? I'm not to familiar with ExceptionMappers.

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
ci:test:search

**Comment by joshuacords** [comment]
ci:test:relevant

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -18033,6 +18033,8 @@ unable-to-find-article-resource-for-article-with-id-x-while-gathering-its-refere
 unable-to-find-x=Unable to find {0}.
 unable-to-import-because-the-blueprint-configuration-is-invalid=Unable to import because the blueprint configuration is invalid.
 unable-to-import-because-the-element-configuration-is-invalid=Unable to import because the element configuration is invalid.
+unable-to-import-element-with-the-same-external-reference-code-with-the-same-as-an-existing-blueprint=Unable to import element with the same external reference code as an existing blueprint.
```
Comment (thektan) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:18036]: Small typo.

```suggestion
unable-to-import-blueprint-with-the-same-external-reference-code-with-the-same-as-an-existing-blueprint=Unable to import blueprint with the same external reference code as an existing blueprint.
```

**Review by thektan — CHANGES_REQUESTED** [review]
Just noticed a small typo. Frontend changes look good 👍

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
> Hi @gustavolimav, this looks pretty good - could you explain how the DuplicateSXPBlueprintExternalReferenceCodeException gets checked for and then thrown? I'm not to familiar with ExceptionMappers.

The Exception throws [here](https://github.com/liferay/liferay-portal/blob/6f0f6d0eff98062915dc45990f1847d8d2ec0447/modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/service/impl/SXPElementLocalServiceImpl.java#L202) and because it is registred in OSGI with `service = ExceptionMapper.class` when some Exception throws it will check if the type is registered and call the ExceptionMapper file. I just needed to implement `getProblem` to return the message and status that I want, and implement `isSanitize` to avoid the base class sanitizing it and remove the title from the response.

Other details about EceptionMappers I read here (Actually I had forgot to put `@Provider`):
https://www.ibm.com/docs/en/was-nd/8.5.5?topic=applications-defining-exception-mappers-resource-exceptions-errors
https://dennis-xlc.gitbooks.io/restful-java-with-jax-rs-2-0-2rd-edition/content/en/part1/chapter7/exception_handling.html

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by joshchong** [comment]
Sent to BChan:
https://github.com/brianchandotcom/liferay-portal/pull/130108

**Pull request closed without merging**